packages/layout/src/tree-layout/drawing-manager/drawingManager.js

```diff
@@ -0,0 +1,127 @@
+    // resolve alingment
+    /**
+     * alignwith :
+     * alignment : left | right | h-center | v-center |default
+     * logic if Alignment is need search the div ,
+     * create a child div according to alignment,
+     * set the measurements and append it as child,
+     * (check if alignment possible)
+     * replace node id of parent with child
+     */
+        this._resolveAligment(this.data);
+        this._drawComponent(this.data);
+    }
+
+    _resolveAligment (componentData) {
+        componentData.children.forEach((component) => {
+            if (component.model && component.model.host && component.model.host.alignWith) {
+                let childNode;
+                const node = this._findNode(component._id);
+                const refNode = this._findNode(this.componentMap.get(component.model.host.alignWith).renderAt);
+                switch (component.model.host.alignment) {
+                case 'left':
+                    childNode = this._getChildNode(node.top,
+              refNode.left,
+              node.height,
+              Math.abs(node.width - Math.abs(refNode.left - node.left)),
+              node._id);
+                    break;
+                case 'right':
+                    childNode = this._getChildNode(node.top,
+              node.left,
+              node.height,
+              Math.abs(node.width - Math.abs(node.left + node.width - (refNode.left + refNode.width))),
+              node._id);
+                    break;
+                case 'top':
+                    childNode = this._getChildNode(refNode.top,
+              node.left,
+              Math.abs(node.height - Math.abs(refNode.top - node.top)),
+              node.width,
+              node._id);
+                    break;
+                case 'bottom':
+                    childNode = this._getChildNode(node.top,
+              node.left,
+              Math.abs(node.top - refNode.top + refNode.height),
+              node.width,
+              node._id);
+                    break;
+                case 'h-center':
+                    childNode = this._getChildNode(node.top,
+              refNode.left,
+              node.height,
+              refNode.width,
+              node._id);
+                    break;
+                case 'v-center':
+                    childNode = this._getChildNode(refNode.top,
+              node.left,
+              refNode.height,
+              node.width,
+              node._id);
+                    break;
+                default:
+                    break;
+                }
+        // check if model in parent component
+                this.componentMap.get(component.model.host.componentName).renderAt = `${component._id}-holder`;
+                this.componentRenderer.parentDiv.appendChild(childNode);
+            }
+            this._resolveAligment(component);
+        });
+    }
+
+    _getChildNode (top, left, height, width, _id) {
+        const childNodeDim = {};
```

Comment:
can be written as 
const childNodeDim = {
top,
left,
height,
width,
_id: `${_id}-holder`
}

Also, why is _ required?